pyxrf/model/fileio.py

```diff
@@ -453,12 +455,14 @@ def delete_points(self):
     def get_sum(self, mask=None):
         if len(self.point1) == 0 and len(self.point2) == 0:
             SC = SpectrumCalculator(self.raw_data)
-            return SC.get_spectrum(mask=mask)
+            spec = SC.get_spectrum(mask=mask)
         else:
             SC = SpectrumCalculator(self.raw_data,
                                     pos1=self.point1,
                                     pos2=self.point2)
-            return SC.get_spectrum()
+            spec = SC.get_spectrum()
+        # Return the 'sum' spectrum as regular 64-bit float (raw data is in 'np.float32')
```

Comment:
Why does it have to be different from the dtype of the raw data?

Reply:
Averaged spectrum (sum of all spectra) is small (4096 points, not 1000x1000x4096 points as for raw data), may cover wide range of values and is used for nonlinear fitting (to adjust a lot of model parameters, not just amplitudes). So working with 64-bit floats is justified. Also all computations in PyXRF are traditionally done using 64-bit floats and processed maps are saved as 64-bit floats. It is unreasonable to store raw data, which are essentially integers in limited range, as 64-bit numbers. Changing the precision to 32 bits saves 1/2 memory.

Reply:
Thanks for the explanation. What about using the uint32 or uint64 for the raw data instead?

Reply:
It seems that using floats is more universal. PyXRF can work with floating point raw data, it's just that the detectors used on the beamlines are producing integers. Actually float32 is still very precise representation.

Reply:
Also, int16 is insufficient.

Reply:
OK, good to know.

Reply:
> Actually float32 is still very precise representation.

That's right. In ptycho GUI we have a switch that now defaults to float32 to save memory usage.